lib/trie/trie.go

```diff
@@ -26,6 +26,11 @@ type Trie struct {
 	// pruner to detect with database keys (trie node Merkle values) can
 	// be deleted.
 	deletedMerkleValues map[string]struct{}
+	// newDeletedMerkleValues contains deleted Merkle values for operations
+	// in progress. Its goal is to prevent having the trie `deletedMerkleValues`
+	// in a bad state if one of the trie operation modified some (deep copied) trie
+	// node and then failed.
+	newDeletedMerkleValues map[string]struct{}
```

Comment:
nit: maybe change this to something like pendingDeletedMerkleValues or something like this?

Reply:
Done